### PR TITLE
Direct Support for Persistent Indexes

### PR DESCRIPTION
**Pull request by iyobo**
This is just an enhancement to ArangoJS to allow it handle persistent indexes in a non-arbitrary manner i.e. besides just collection.createIndex.

Users can now ensure Persistent indexes (Arango 3+) using collection.createPersistentIndex(fields, opts).

Similar to skiplists.

### CODE REVIEW

**Comment by pluma** [comment]
Thank you for the PR. For legal reasons we need a signed Contributor License Agreement (CLA) before we can merge your code into the project.

You can download the form from our website: https://www.arangodb.com/documents/cla.pdf

Once you've filled out and signed it, please send it to us via e-mail: cla@arangodb.com or fax: +49-221-2722999-88.

**Comment by pluma** [comment]
We haven't received a CLA yet so I can't merge this PR. Feel free to re-open the PR at any point.

**Comment by iyobo** [comment]
Signed

**Comment by pluma** [comment]
Awesome! Thanks and again apologies for the long wait.

Could you add a section about the method to the API documentation in the README, though? It should probably go near the other indexes.

I'll merge the PR once the CLA has been confirmed.

File: package.json [hunk 1]
```diff
@@ -50,11 +50,14 @@
   },
   "devDependencies": {
     "babel": "6.5.2",
-    "babel-cli": "6.18.0",
-    "babel-core": "6.18.0",
+    "babel-cli": "^6.18.0",
+    "babel-core": "^6.18.0",
```
Comment (pluma) [inline package.json]: What's going on here? The devDependencies are managed by greenkeeper and should be locked to the latest version already.

File: package.json [hunk 2]
```diff
@@ -50,11 +50,14 @@
...
     "babel-loader": "6.2.6",
     "babel-plugin-add-module-exports": "0.2.1",
+    "babel-plugin-check-es2015-constants": "^6.8.0",
     "babel-plugin-transform-builtin-extend": "1.1.0",
+    "babel-plugin-transform-es2015-arrow-functions": "^6.8.0",
+    "babel-plugin-transform-es2015-block-scoping": "^6.18.0",
```
Comment (pluma) [inline package.json]: These seem to be included in `babel-preset-es2015`. Why are you adding them explicitly?

**Comment by pluma** [comment]
@iyobo Can you clarify the changes I pointed out?

**Comment by iyobo** [comment]
I have addressed all concerns.

**Comment by pluma** [comment]
Hi @iyobo,

thanks for the changes.

Could you please remove the `.idea` folder? You can add it to `.gitignore` if that helps.

**Comment by iyobo** [comment]
Done

**Comment by pluma** [comment]
@iyobo You've added the folder to `.gitignore` but you didn't remove the actual folder from git. Try `git rm .idea`.

**Comment by pluma** [comment]
@iyobo Could you please remove .idea/workspace.xml from the repository so this can be merged?

**Comment by pluma** [comment]
Thank you for your PR. I didn't notice I had commit access to your fork.

**Comment by pluma** [comment]
The changes are now in v5.5.0.

**Pull request merged**